txircd/ircd.py

```diff
@@ -385,7 +375,150 @@ def reloadModule(self, moduleName):
+			config["server_id"] += randFromName.choice(string.digits + string.uppercase)
+			config["server_id"] += randFromName.choice(string.digits + string.uppercase)
+		if len(config["server_id"]) != 3 or not config["server_id"].isalnum() or not config["server_id"][0].isdigit():
+			raise ConfigValidationError("server_id", "value must be a 3-character alphanumeric string starting with a number.")
+		if "server_description" not in config:
+			raise ConfigValidationError("server_description", "required item not found in configuration file.")
+		if not isinstance(config["server_description"], basestring):
+			raise ConfigValidationError("server_description", "value must be a string")
+		if "network_name" not in config:
+			raise ConfigValidationError("network_name", "required item not found in configuration file.")
+		if not isinstance(config["network_name"], basestring):
+			raise ConfigValidationError("network_name", "value must be a string")
+		if len(config["network_name"]) > 32:
+			config["network_name"] = config["network_name"][:32]
+			self.logConfigValidationWarning("network_name", "value is too long", config["network_name"])
+		if not "bind_client" not in config:
+			config["bind_client"] = [ "tcp:6667:interface={::}" ]
+			self.logConfigValidationWarning("bind_client", "no default client binding specified", "[ \"tcp:6667:interface={::}\" ]")
+		if not isinstance(config["bind_client"], list):
+			raise ConfigValidationError("bind_client", "value must be a list")
+		for bindDesc in config["bind_client"]:
+			if not isinstance(bindDesc, basestring):
+				raise ConfigValidationError("bind_client", "every entry must be a string")
+		if "bind_server" not in config:
+			config["bind_server"] = []
+		if not isinstance(config["bind_server"], list):
+			raise ConfigValidationError("bind_server", "value must be a list")
+		for bindDesc in config["bind_server"]:
+			if not isinstance(bindDesc, basestring):
+				raise ConfigValidationError("bind_server", "every entry must be a string")
+		if "modules" not in config:
+			config["modules"] = []
+		if not isinstance(config["modules"], list):
+			raise ConfigValidationError("modules", "value must be a list")
+		for module in config["modules"]:
+			if not isinstance(module, basestring):
+				raise ConfigValidationError("modules", "every entry must be a string")
+		if "links" in config:
+			if not isinstance(config["links"], dict):
+				raise ConfigValidationError("links", "value must be a dictionary")
+			for desc, server in config["links"].iteritems():
+				if not isinstance(desc, basestring):
+					raise ConfigValidationError("links", "\"{}\" is an invalid server description".format(desc))
+				if not isinstance(server, dict):
+					raise ConfigValidationError("links", "values for \"{}\" must be a dictionary".format(desc))
+				if "connect_descriptor" not in server:
+					raise ConfigValidationError("links", "server \"{}\" must contain a \"connect_descriptor\" value".format(desc))
```

Comment:
If we're checking this here, we should be able to remove the check from `connectServer`.